watcher_test.go

```diff
@@ -82,21 +82,26 @@ func testCancelByFunc(t *testing.T, fn func(wg *sync.WaitGroup, loader *Loader,
 	assert.Equal(t, 0, countWatchers(loader))
 }
 
-func TestWatch_Many(t *testing.T) {
+func TestWatchMany(t *testing.T) {
```

Comment:
tests with golang v1.22 where timing out. it would never close.

Reply:
What's the reason for this?

Reply:
Let's fix it by disposing in `Close()`:

```go
func (w *watcher) Close() error {
	w.changeState(isRunning, isCanceled)
	w.dispose()
	return nil
}
```

Reply:
Fixed !